lib/streamlit/__init__.py

```diff
@@ -194,261 +185,24 @@ def _update_logger() -> None:
-    you would do the following:
-
-    >>> st.experimental_set_query_params(
-    ...     show_map=True,
-    ...     selected=["asia", "america"],
-    ... )
-
-    """
-    ctx = _get_script_run_ctx()
-    if ctx is None:
-        return
-    ctx.query_string = _parse.urlencode(query_params, doseq=True)
-    msg = _ForwardMsg_pb2.ForwardMsg()
-    msg.page_info_changed.query_string = ctx.query_string
-    ctx.enqueue(msg)
-
-
-@_contextlib.contextmanager
-def spinner(text: str = "In progress...") -> Iterator[None]:
-    """Temporarily displays a message while executing a block of code.
-
-    Parameters
-    ----------
-    text : str
-        A message to display while executing that block
-
-    Example
-    -------
-
-    >>> with st.spinner('Wait for it...'):
-    >>>     time.sleep(5)
-    >>> st.success('Done!')
-
-    """
-    import streamlit.runtime.legacy_caching.caching as legacy_caching
-    import streamlit.runtime.caching as caching
-    from streamlit.proto.Spinner_pb2 import Spinner as SpinnerProto
-    from streamlit.string_util import clean_text
-
-    # @st.cache optionally uses spinner for long-running computations.
-    # Normally, streamlit warns the user when they call st functions
-    # from within an @st.cache'd function. But we do *not* want to show
-    # these warnings for spinner's message, so we create and mutate this
-    # message delta within the "suppress_cached_st_function_warning"
-    # context.
-    with legacy_caching.suppress_cached_st_function_warning():
-        with caching.suppress_cached_st_function_warning():
-            message = empty()
-
-    try:
-        # Set the message 0.1 seconds in the future to avoid annoying
-        # flickering if this spinner runs too quickly.
-        DELAY_SECS = 0.1
-        display_message = True
-        display_message_lock = _threading.Lock()
-
-        def set_message():
-            with display_message_lock:
-                if display_message:
-                    with legacy_caching.suppress_cached_st_function_warning():
-                        with caching.suppress_cached_st_function_warning():
-                            spinner_proto = SpinnerProto()
-                            spinner_proto.text = clean_text(text)
-                            message._enqueue("spinner", spinner_proto)
-
-        _add_script_run_ctx(_threading.Timer(DELAY_SECS, set_message)).start()
-
-        # Yield control back to the context.
-        yield
-    finally:
-        if display_message_lock:
-            with display_message_lock:
-                display_message = False
-        with legacy_caching.suppress_cached_st_function_warning():
-            with caching.suppress_cached_st_function_warning():
-                message.empty()
+# Experimental APIs
+experimental_user = _UserInfoProxy()
+experimental_singleton = _singleton
+experimental_memo = _memo
+experimental_get_query_params = _gather_metrics(_get_query_params)
```

Comment:
One issue here is that the function name is not matching the old name: Calls to this function would be tracked via the name`_get_query_params`, compared to `experimental_get_query_params` in the current release. I started to solve this via a mapping here:

https://github.com/streamlit/streamlit/blob/0496d3d066f69cefe295d4925ed8f1875723ba99/lib/streamlit/runtime/metrics_util.py#L50

But this is a bit ugly, and I think a better long-term solution is to add a `name` parameter to `gather_metrics` that allows to overwrite the name that is used for tracking calls to this function. I will look into this tomorrow.

Reply:
To not block this PR, it would be good to add the following mappings to `_CALLABLE_NAME_MAPPING`:
 
```python
"_get_query_params": "experimental_get_query_params",
"_set_query_params": "experimental_set_query_params",
"_show": "experimental_show",
"_set_option": "set_option",
```

Reply:
Done!